src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/service/ActionService.java

```diff
@@ -0,0 +1,37 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowmonitoring.service;
+
+import org.openkilda.server42.messaging.FlowDirection;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ActionService {
+
+    /**
+     * Check flow SLA is violated.
+     */
+    public void checkFlowSla(String flowId, FlowDirection direction, long latency, long maxLatency) {
+        if (latency > maxLatency) {
+            long latencyMs = latency / 1000;
+            long maxLatencyMs = maxLatency / 1000;
+            log.warn("{} path for flow '{}' violates latency SLA {}/{} ms",
```

Comment:
maybe it would be better to have adaptive logging here, I would prefer to have a warn once in a minute at least, and all other triggers to be info for example

Reply:
It's for testing purpose. This section will be covered in next patches on flow SLA reactions.